portlets/opencps-portlet/docroot/WEB-INF/src/org/opencps/accountmgt/service/persistence/BusinessFinderImpl.java

```diff
@@ -0,0 +1,296 @@
+
+		if (Validator.isNotNull(keywords)) {
+			names = CustomSQLUtil.keywords(keywords);
+		} else {
+			andOperator = true;
+		}
+
+		return _countBussiness(groupId, names, accountStatus, businessDomain,
+				andOperator);
+	}
+
+	private int _countBussiness(long groupId, String[] keywords,
+			int accountStatus, String businessDomain, boolean andOperator) {
+
+		/*
+		 * keywords = CustomSQLUtil .keywords(keywords, false);
+		 */
+
+		Session session = null;
+		try {
+			session = openSession();
+			String sql = CustomSQLUtil.get(COUNT_BUSINESS);
+			if (keywords != null && keywords.length > 0) {
+				sql = CustomSQLUtil.replaceKeywords(sql,
+						"lower(opencps_acc_business.name)", StringPool.LIKE,
+						true, keywords);
+
+				sql = CustomSQLUtil.replaceKeywords(sql,
+						"lower(opencps_acc_business.shortName))",
+						StringPool.LIKE, true, keywords);
+
+				sql = CustomSQLUtil.replaceKeywords(sql,
+						"lower(opencps_acc_business.enName", StringPool.LIKE,
+						true, keywords);
+			} else {
+				sql = StringUtil
+						.replace(
+								sql,
+								"AND ((lower(opencps_acc_business.name) LIKE ? [$AND_OR_NULL_CHECK$])",
+								StringPool.BLANK);
+
+				sql = StringUtil
+						.replace(
+								sql,
+								"OR (lower(opencps_acc_business.shortName) LIKE ? [$AND_OR_NULL_CHECK$])",
+								StringPool.BLANK);
+
+				sql = StringUtil
+						.replace(
+								sql,
+								"OR (lower(opencps_acc_business.enName) LIKE ? [$AND_OR_NULL_CHECK$]))",
+								StringPool.BLANK);
+			}
+			
+			if (Validator.isNull(businessDomain)) {
+				sql = StringUtil
+						.replace(
+								sql,
+								"AND (opencps_acc_businessdomain.businessDomainCode = ?)",
+								StringPool.BLANK);
+				sql = StringUtil.replace(sql, 
+						"INNER JOIN opencps_acc_businessdomain ON opencps_acc_business.businessId = opencps_acc_businessdomain.businessId"
+						, StringPool.BLANK);
+
+			}
+
+			if (accountStatus == -1) {
+				sql = StringUtil.replace(sql,
+						"AND (opencps_acc_business.accountStatus = ?)",
+						StringPool.BLANK);
+			}
+			sql = CustomSQLUtil.replaceAndOperator(sql, andOperator);
+			SQLQuery q = session.createSQLQuery(sql);
+			q.setCacheable(false);
+
+			q.addScalar(COUNT_COLUMN_NAME, Type.INTEGER);
+
+			QueryPos qPos = QueryPos.getInstance(q);
+
+			qPos.add(groupId);
+
+			if (accountStatus != -1) {
+				qPos.add(accountStatus);
+			}
+
+			if (Validator.isNotNull(businessDomain)) {
+				qPos.add(businessDomain);
+			}
+
+			if (keywords != null && keywords.length > 0) {
+				
+				qPos.add(keywords, 2);
+				
+				qPos.add(keywords, 2);
+
+				qPos.add(keywords, 2);
+
+			}
+			Iterator<Integer> itr = q.iterate();
+
+			if (itr.hasNext()) {
+				Integer count = itr.next();
+
+				if (count != null) {
+					return count.intValue();
+				}
+			}
+
+		} catch (Exception e) {
+			_log.error(e);
+		} finally {
+			session.close();
+		}
+
+		return 0;
+	}
+
+	private Log _log = LogFactoryUtil.getLog(BusinessFinderImpl.class);
```

Comment:
@huymq phần này sẽ không đặt try catch ở đây, tầng này để throw ra exception.

Nhiều phần viết ntn, sửa lại toàn bộ nhé.